src/main/java/org/reaktivity/nukleus/mqtt/internal/stream/MqttServerFactory.java

```diff
@@ -265,12 +300,299 @@ private RouteFW resolveTarget(
+        long routeId,
+        long streamId,
+        long traceId,
+        long budgetId,
+        int initialPadding,
+        int initialCredit)
+    {
+        final WindowFW window = windowRW.wrap(writeBuffer, 0, writeBuffer.capacity())
+                                        .routeId(routeId)
+                                        .streamId(streamId)
+                                        .traceId(traceId)
+                                        .budgetId(budgetId)
+                                        .credit(initialCredit)
+                                        .padding(initialPadding)
+                                        .build();
+
+        receiver.accept(window.typeId(), window.buffer(), window.offset(), window.sizeof());
+    }
+
+    private void doReset(
+        MessageConsumer receiver,
+        long routeId,
+        long streamId,
+        long traceId,
+        long authorization)
+    {
+        final ResetFW reset = resetRW.wrap(writeBuffer, 0, writeBuffer.capacity()).routeId(routeId)
+                                     .streamId(streamId)
+                                     .traceId(traceId)
+                                     .authorization(authorization)
+                                     .build();
+
+        receiver.accept(reset.typeId(), reset.buffer(), reset.offset(), reset.sizeof());
+    }
+
+    private void doSignal(
+        MessageConsumer receiver,
+        long routeId,
+        long streamId,
+        long traceId)
+    {
+        final SignalFW signal = signalRW.wrap(writeBuffer, 0, writeBuffer.capacity()).routeId(routeId)
+                                        .streamId(streamId)
+                                        .traceId(traceId)
+                                        .build();
+
+        receiver.accept(signal.typeId(), signal.buffer(), signal.offset(), signal.sizeof());
+    }
+
+    private int decodePacketType(
+        MqttServer server,
+        final long traceId,
+        final long authorization,
+        final long budgetId,
+        final DirectBuffer buffer,
+        final int offset,
+        final int limit)
+    {
+        final MqttPacketFixedHeaderFW packet = mqttPacketFixedHeaderRO.tryWrap(buffer, offset, limit);
+
+        if (packet != null)
+        {
+            final int length = packet.remainingLength();
+            final MqttPacketType packetType = MqttPacketType.valueOf(packet.typeAndFlags() >> 4);
+            final MqttServerDecoder decoder = decodersByPacketType.getOrDefault(packetType, decodeIgnoreOne);
+
+            if (limit - packet.limit() >= length)
+            {
+                server.decoder = decoder;
+            }
+        }
+        return offset;
+    }
+
+    private int decodeConnect(
+        MqttServer server,
+        final long traceId,
+        final long authorization,
+        final long budgetId,
+        final DirectBuffer buffer,
+        final int offset,
+        final int limit)
+    {
```

Comment:
Check protocol correctness in `decodeXXX` methods, potentially rejecting the frame and forcing cleanup if protocol is malformed in some way. For example, detecting a second connect when already connected should happen here.

But defer logical processing of correct protocol use to `server.onDecodeXXX` methods.